src/auth/permissions/__tests__/index.test.ts

```diff
@@ -0,0 +1,187 @@
+      }
+    });
+    test('ERROR - Cannot grant a permission that doesn\'t already exist', async() => {
+      expect.assertions(1);
+      try {
+        const query = await permissionsInterface.grantExisting({
+          resource: 'constants',
+          access:'read',
+          permissionLevel: 'parent',
+          role:'VISITOR' });
+      } catch (error) {
+        expect(error.message).toBe(
+          'Permission entry or role does not exist, please use grantNew method');
+      }
+    });
+    test('ERROR - Duplicate link between role and permission entry', async() => {
+      expect.assertions(1);
+      try {
+        const query = await permissionsInterface.grantExisting({
+          resource: 'constants',
+          access:'read',
+          permissionLevel: 'own',
+          role:'VISITOR' });
+      } catch (error) {
+        expect(error.message).toBe('Permission entry is already associated to this role');
+      }
+    });
+  });
+
+  describe('::revoke', () => {
+    test('SUCCESS - deletes existing link between a permission entry and role', async() => {
+      expect.assertions(1);
+      try {
+        const query = await permissionsInterface.revoke({
+          resource: 'constants',
+          access:'read',
+          permissionLevel: 'own',
+          role:'VISITOR' });
+        expect(query).toBe(1);
+      } catch (error) {
+        expect(error).toBeFalsy();
+      }
+    });
+
+    test('ERROR - cannot delete when permission entry and role are not linked', async() => {
+      expect.assertions(1);
+      try {
+        const query = await permissionsInterface.revoke({
+          resource: 'constants',
+          access:'read',
+          permissionLevel: 'own',
+          role:'VOLUNTEER' });
+        expect(query).toBe(false);
+      } catch (error) {
+        expect(error.message).toBe('Permission entry is not linked to role');
+      }
+    });
+
+    test('ERROR - cannot delete when permission entry does not exist', async() => {
+      expect.assertions(1);
+      try {
+        const query = await permissionsInterface.revoke({
+          resource: 'organisations_details',
+          access:'read',
+          permissionLevel: 'child',
+          role:'VOLUNTEER' });
+      } catch (error) {
+        expect(error.message).toBe('Permission entry does not exist');
+      }
+    });
+  });
+
+  describe('::roleHas', () => {
+    test('SUCCESS - returns false for non matching permissions & user', async() => {
+      try {
+        const query = await permissionsInterface.roleHas({
+          resource: 'constants',
+          access:'read',
+          permissionLevel: 'own',
+          role:'VOLUNTEER' });
+        expect(query.exists).toBe(false);
+      } catch (error) {
+        expect(error).toBeFalsy();
+      }
+    });
+    test('SUCCESS - returns true for matching permissions & user', async() => {
+      try {
+        const query = await permissionsInterface.roleHas({
+          resource: 'constants',
```

Comment:
I think all these literal string values that are being passing as arguments should actually be constants that are imported from the permissions module. That's how it is in the initial design. It decouples meaning from value.